atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/memory/InMemoryAtlasDbFactory.java

```diff
@@ -68,6 +69,8 @@ public String getType() {
 
     @Override
     public InMemoryKeyValueService createRawKeyValueService(KeyValueServiceConfig config) {
+        AtlasDbVersion.ensureVersionReported();
```

Comment:
I actually don't know what the right answer here is. The docs from palantir.github.io don't have many examples on what the right pattern for constructing an Atlas connection is, so I'm not certain how most places go about doing this. I would prefer if there was a single path which would cover everything, but I couldn't find that path. Maybe you have a suggestion?